### PR TITLE
Dev team 0 sp 2 jwt security implementation

### PR DESCRIPTION
**Pull request by liza-jahan**

### CODE REVIEW

File: src/main/java/com/lina/airline/auth/UserNamePwdAuthenticationProvider.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import org.springframework.stereotype.Component;
+
+@AllArgsConstructor
+@Component
+
+public class UserNamePwdAuthenticationProvider implements AuthenticationProvider {
+    private final PasswordEncoder passwordEncoder;
+    private final UserDetailsService userDetailsService;
+
+//    @Override
```
Comment (rktirtho) [inline src/main/java/com/lina/airline/auth/UserNamePwdAuthenticationProvider.java:21]: Remove commented code

File: src/main/java/com/lina/airline/controller/AuthenticationController.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    private String jwtSecretValue;
+
+    @PostMapping("/login")
+    public ResponseEntity<AuthenticationResponse> login(@RequestBody AuthenticationRequest authRequest) {
+
+        String jwt = "";
+        Authentication authentication = UsernamePasswordAuthenticationToken.unauthenticated(authRequest.email(),
+                authRequest.password());
+        Authentication authenticationResponse = authenticationManager.authenticate(authentication);
+        if (authenticationResponse != null && authenticationResponse.isAuthenticated()) {
```
Comment (rktirtho) [inline src/main/java/com/lina/airline/controller/AuthenticationController.java:40]: Business Logic should not be added to the controller layer. Move it to the service layer. And also create a new Service for the JWT token generator.

File: src/main/java/com/lina/airline/controller/AuthenticationController.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        }
+        return ResponseEntity.status(HttpStatus.OK).header(ApplicationConstantsUtils.JWT_HEADER, jwt).body(new AuthenticationResponse(HttpStatus.OK.getReasonPhrase(), jwt));
+    }
+
+}
+
+
+//
+//
+//    HttpHeaders headers = new HttpHeaders();
```
Comment (rktirtho) [inline src/main/java/com/lina/airline/controller/AuthenticationController.java:60]: Remove commented code

File: src/main/java/com/lina/airline/controller/UserController.java [hunk 4]
```diff
@@ -18,15 +18,23 @@
...
 @RequiredArgsConstructor
 public class UserController {
 
    public final UserService userService;
 
-    @PostMapping("/")
-    public UUID registrationUser(@RequestBody @Valid UserRegistrationRequest userRegistrationRequest){
-        return userService.saveUser(userRegistrationRequest);
+    @PostMapping("/register")
+    public ResponseEntity<APIResponse<CreationResponse>> registrationUser(@RequestBody @Valid UserRegistrationRequest userRegistrationRequest){
```
Comment (rktirtho) [inline src/main/java/com/lina/airline/controller/UserController.java:28]: This is a completely different task. Never do multiple task in same commit. It is OK for now

File: src/main/java/com/lina/airline/dto/response/AuthenticationResponse.java [hunk 5]
```diff
@@ -0,0 +1,4 @@
+package com.lina.airline.dto.response;
+
+public record AuthenticationResponse ( String status, String token){
```
Comment (rktirtho) [inline src/main/java/com/lina/airline/dto/response/AuthenticationResponse.java:3]: Do you know about Java record?

File: src/main/java/com/lina/airline/entity/RoleEntity.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+import jakarta.persistence.*;
+import lombok.*;
+import org.hibernate.annotations.GenericGenerator;
+
+import java.util.HashSet;
+import java.util.Set;
+@Getter
+@Setter
+@Builder
```
Comment (rktirtho) [inline src/main/java/com/lina/airline/entity/RoleEntity.java:11]: Do you know if a builder pattern is required there?

File: src/main/java/com/lina/airline/filter/JwtFilter.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+@RequiredArgsConstructor
+@Component
+@AllArgsConstructor
+public class JwtFilter extends OncePerRequestFilter {
+    private CustomUserDetailsServiceImp userDetailsService;
+    private JWTServiceImp jwtService;
+    ApplicationContext context;
+
+    public JwtFilter(JWTServiceImp jwtService) {
```
Comment (rktirtho) [inline src/main/java/com/lina/airline/filter/JwtFilter.java:29]: Already annotated with RequiredArgsConstructor and AllArgsContructor. This constructor should not declare again. But You may get a issue at there. B'coz you have to declare the jwtTokenService as final

File: src/main/java/com/lina/airline/utils/ApplicationConstantsUtils.java [hunk 8]
```diff
@@ -0,0 +1,7 @@
+package com.lina.airline.utils;
+
+public final class ApplicationConstantsUtils {
+    public  static final  String JWT_SECRET_KEY="JWT_SECRET";
```
Comment (rktirtho) [inline src/main/java/com/lina/airline/utils/ApplicationConstantsUtils.java:4]: This properties should takes from the application properties file. This value may changes based on environment.

File: src/main/java/com/lina/airline/service/impl/JWTServiceImp.java [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+@Service
+public class JWTServiceImp {
+    private String secretkey = "";
```
Comment (rktirtho) [inline src/main/java/com/lina/airline/service/impl/JWTServiceImp.java:21]: Do we need to add secrateKey variable in the object properties?

**Review by rktirtho — COMMENTED** [review]
Check the comment and incorporate the review feedback. I have also added some queries. Please answer them in the reply.

File: src/main/java/com/lina/airline/dto/response/AuthenticationResponse.java (hunk 5 above)
Comment (liza-jahan) [reply src/main/java/com/lina/airline/dto/response/AuthenticationResponse.java:3]: yes.i know it.

File: src/main/java/com/lina/airline/filter/JwtFilter.java (hunk 7 above)
Comment (liza-jahan) [reply src/main/java/com/lina/airline/filter/JwtFilter.java:29]: Fixed this issue.

File: src/main/java/com/lina/airline/auth/UserNamePwdAuthenticationProvider.java (hunk 1 above)
Comment (liza-jahan) [reply src/main/java/com/lina/airline/auth/UserNamePwdAuthenticationProvider.java:21]: Done

File: src/main/java/com/lina/airline/controller/AuthenticationController.java (hunk 3 above)
Comment (liza-jahan) [reply src/main/java/com/lina/airline/controller/AuthenticationController.java:60]: Done it

**Pull request merged**